tests/unit/mcpgateway/plugins/framework/external/mcp/test_client_config.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+"""Location: ./tests/unit/mcpgateway/plugins/framework/external/mcp/test_client_config.py
+Copyright 2025
+SPDX-License-Identifier: Apache-2.0
+Authors: Fred Araujo
+
+Additional unit tests for ExternalPlugin client.
+Tests for error conditions, edge cases, and uncovered code paths.
+"""
+
+import os
+import pytest
+from unittest.mock import AsyncMock, Mock, patch
+
+from mcp.types import CallToolResult
+
+from mcpgateway.models import Message, PromptResult, ResourceContent, Role, TextContent
+from mcpgateway.plugins.framework.external.mcp.client import ExternalPlugin
+from mcpgateway.plugins.framework.models import (
```

Comment:
Should we start importing everything from `mcpgateway.plugins.framework`?   I think the test cases right now are the ones that don't follow this convention but we could start doing it as folks look for examples.